packages/core/core/src/utils/context/index.ts

```diff
@@ -28,9 +28,15 @@ const applyContextToApi = (
   hooks: ApplyingContextHooks = { before: NOP, after: NOP }
 ) =>
   Object.entries(api)
-    .reduce((prev, [key, fn]: any) => ({
+    .reduce((prev, [callName, fn]: any) => ({
       ...prev,
-      [key]: async (...args) => hooks.after(await fn(context, ...hooks.before(args)))
+      [callName]: async (...args) => {
+        const transformedArgs = hooks.before({ callName, args });
+        const response = await fn(context, ...transformedArgs);
+        const transformedResponse = hooks.after({ callName, args, response });
+
```

Comment:
Wdyt about adding there Logger?

Reply:
logging is something that I'll be considering in the new issue: what we want to log, what verbosity etc.